examples/src/custom_mpl_cmm_example.py

```diff
@@ -0,0 +1,114 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Example to create a custom implementation of the ESDK-MPL ICryptographicMaterialsManager class."""
+
+from aws_cryptographic_materialproviders.mpl import AwsCryptographicMaterialProviders
+from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+from aws_cryptographic_materialproviders.mpl.models import (
+    CreateDefaultCryptographicMaterialsManagerInput,
+    SignatureAlgorithmNone,
+)
+from aws_cryptographic_materialproviders.mpl.references import ICryptographicMaterialsManager, IKeyring
+
+import aws_encryption_sdk
+from aws_encryption_sdk import CommitmentPolicy
+
+
+# Custom CMM implementation using the MPL.
+# This CMM only allows encryption/decryption using signing algorithms.
+# It wraps an underlying CMM implementation and checks its materials
+# to ensure that it is only using signed encryption algorithms.
+class MPLCustomSigningSuiteOnlyCMM(ICryptographicMaterialsManager):
+    """Example custom crypto materials manager class."""
+
+    def __init__(self, keyring: IKeyring) -> None:
```

Comment:
Could this optionally take in a CMM, and if none is provided, then construct a default one with the provided keyring?
Consider
```suggestion
    def __init__(self, keyring: IKeyring, cmm: ICryptographicMaterialsManager) -> None:
        if cmm is None:
             ...
```
The underlying CMM can be *any CMM*. That point comes through clearly by letting this custom CMM take in any CMM

Reply:
Sounds good! Thanks.